docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,172 @@
+- This design allows for grouping of metrics for which you would want to define particular customizations. Grouping allows hiveconfig to be parsable, but implementation of this model would require an efficient solution to validate there are no ambiguous entries for a metric, and storing it within the code for parsing purposes.
+- There needs to be clear logic defining which metrics can have `minimumDuration` versus which ones can't. Perhaps if a metric has `seconds` in the name can be the decider.
+- Current configuration of using a shorthand camelCase key to refer to a corresponding metric while configuring hiveconfig.spec.metricsConfig.metricsWithDuration will be removed, and users will have to use the full metric name in the proposed configuration.
+- Presently, the metrics that support hiveconfig.spec.metricsConfig.metricsWithDuration are optional, so they will not be logged unless they're specified in HiveConfig. After this redesign, the metrics that are currently optional will continue to be optional - that behaviour will not change.
+  For example, after the redesign, hive_cluster_deployments_running_transition_seconds will be reported if and only if metricsConfig.clusterDeploymentRelatedMetrics.metrics[hive_cluster_deployments_running_transition_seconds].minimumDuration is not empty, however hive_cluster_deployment_install_job_duration_seconds, which is a non-optional metric that would support minimumDuration, will be reported even if metricsConfig.clusterDeploymentRelatedMetrics.metrics[hive_cluster_deployment_install_job_duration_seconds].minimumDuration is not specified.
+- Additionally, we would be removing the fixed labels that can be propagated from the cluster deployment, i.e.`cluster_type`, `platform` and `version`. This means, if admins want these labels for the metrics, they will have to define them via  metricsConfig.clusterDeploymentRelatedMetrics[].additionalClusterDeploymentLabels. This would only be a breaking change if the admins suddenly cut over from deprecated/old way of metricsConfig to the new clusterDeploymentRelatedMetrics, sufficient documentation should help though.
+
+#### Deprecation
+
+In order to implement this change, we will have to deprecate the existing hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels.
+As a way to handle the deprecation, we will assert that in a hiveconfig, you can either provide the new configurations (aka `clusterDeploymentRelatedMetrics`) or the old way of hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels.
+Note that only the old design will support the mapping of camelCase key to metric for minimumDuration, the new way will require full metric names only.
+
```

Comment:
As mentioned above, since the two modes differ in several default behaviors, we need to say what happens if *neither* config is given in hiveconfig. And per above, my starting bid is "old behavior". I could maybe be talked out of it if there's a compelling argument on the other side. Thoughts?

Reply:
Leaving aside whether we change which metrics are reported (by default), the default labels _are_ different between the old and new, so IMO it's important to call out explicitly which behavior you get if you leave the config empty.